ledger-service/http-json/src/main/scala/com/digitalasset/http/LedgerClientJwt.scala

```diff
@@ -379,4 +381,76 @@ object LedgerClientJwt {
       }
     }
   }
+
+  object LedgerClientRequestTimeLogger {
+    sealed trait RequestLog {
+      def clientClass: Class[_]
+      def name: String
+    }
+
+    case object SubmitAndWaitForTransactionLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[SynchronousCommandClient]
+      override def name = "submitAndWaitForTransaction"
+    }
+
+    case object SubmitAndWaitForTransactionTreeLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[SynchronousCommandClient]
+      override def name = "submitAndWaitForTransactionTree"
+    }
+
+    case object GetLedgerEndLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[TransactionClient]
+      override def name = "getLedgerEnd"
+    }
+
+    case object ListKnownPartiesLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[PartyManagementClient]
+      override def name = "listKnownParties"
+    }
+
+    case object GetPartiesLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[PartyManagementClient]
+      override def name = "getParties"
+    }
+
+    case object AllocatePartyLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[PartyManagementClient]
+      override def name = "allocateParty"
+    }
+
+    case object ListPackagesLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[PackageClient]
+      override def name = "listPackages"
+    }
+
+    case object GetPackageLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[PackageClient]
+      override def name = "getPackages"
+    }
+
+    case object UploadDarFileLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[PackageManagementClient]
+      override def name = "uploadDarFile"
+    }
+
+    case object GetMeteringReportLog extends RequestLog {
+      override def clientClass: Class[_] = classOf[MeteringReportClient]
+      override def name = "getMeteringReport"
+    }
+
+    private[LedgerClientJwt] def log[T, C](
+        requestLog: RequestLog
+    )(block: => Future[T])(implicit ec: EC, lc: LoggingContextOf[C]): Future[T] = if (
+      logger.debug.isEnabled
+    ) {
+      val start = System.nanoTime()
+      val futureResult = block
+      futureResult.andThen { case _ =>
+        logger.debug(
+          s"Ledger client request ${requestLog.clientClass.getSimpleName} ${requestLog.name} executed, elapsed time: ${System
```

Comment:
If you only use getSimpleName, you could change RequestLog to have a className val instead.

Reply:
Sure. is className a String type or there is a special type for class name?

Reply:
String is fine